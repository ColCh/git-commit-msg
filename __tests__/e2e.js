/* eslint-disable global-require */
jest.mock('fs');

const fs = require('fs');

const launch = (msg) => {
  return new Promise((resolve) => {
    fs.readFileSync.mockReturnValueOnce(msg);
    fs.writeFileSync.mockImplementationOnce((...[, newMsg]) => {
      resolve(newMsg);
    });
    global.FORCE_CLI_EXECUTE = true;
    global.GIT_COMMIT_MSG_HOOK_SKIP_ADDING_EMOJIS = false;
    global.GIT_COMMIT_MSG_HOOK_SKIP_AUTO_SUGGEST = false;
    require('..');
    // require clean cache
  });
};

const msg = [
  'chore(pkg): my commit message',
  `# Please enter the commit message for your changes. Lines starting`,
  `# with '#' will be ignored, and an empty message aborts the commit.`,
  `#`,
  `# On branch master`,
  `# Your branch is up to date with 'origin/master'.`,
  `#`,
  `# Changes to be committed:`,
  `#	modified:   modified-file.txt`,
  `#	deleted:    deleted-file.txt`,
  `#	new file:   new-file.txt`,
  `#`,
  `# Changes not staged for commit:`,
  `#	deleted:    deleted-not-staged.txt`,
  `#	modified:   modified-not-staged.txt`,
  `#`,
  `# Untracked files:`,
  `#	untracked.txt`,
  `#`,
].join('\n');

describe('e2e test', () => {
  it('should match snapshot', async () => {
    expect(await launch(msg)).toMatchSnapshot();
  });
});
