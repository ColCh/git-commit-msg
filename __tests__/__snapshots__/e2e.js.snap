// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`e2e test should match snapshot 1`] = `
"ðŸ˜’ chore(pkg): my commit message
# Found 3 contexts
#     * modified-file
#     * deleted-file
#     * new-file
# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# On branch master
# Your branch is up to date with 'origin/master'.
#
# Changes to be committed:
#	modified:   modified-file.txt
#	deleted:    deleted-file.txt
#	new file:   new-file.txt
#
# Changes not staged for commit:
#	deleted:    deleted-not-staged.txt
#	modified:   modified-not-staged.txt
#
# Untracked files:
#	untracked.txt
#"
`;
